src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts

```diff
@@ -167,12 +169,25 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
         }
     };
 
+    const setFullSizeWidget = (widgetKey: string) => {
+        state.dashboardWidgetInfoList = state.dashboardWidgetInfoList.map((info) => {
+            const widgetSizes = getWidgetConfig(info.widget_name)?.sizes;
+            if (info.widgetKey === widgetKey && widgetSizes) {
+                return {
+                    ...info,
+                    size: (info.size === WIDGET_SIZE.full) ? (widgetSizes[0] ?? WIDGET_SIZE.md) : WIDGET_SIZE.full,
+                };
+            }
+            return info;
+        });
```

Comment:
suggestion: How about to change only target widget? (Code is just sample..)

```suggestion
        const _dashboardWidgetInfoList = cloneDeep(state.dashboardWidgetInfoList);
        const _targetWidgetIndex = _dashboardWidgetInfoList.findIndex((d) => d.widgetKey === widgetKey);
        if (_targetWidgetIndex > -1) {
            const _targetWidget = _dashboardWidgetInfoList[_targetWidgetIndex];
            const widgetSizes = getWidgetConfig(_targetWidget.widget_name)?.sizes;
            _dashboardWidgetInfoList[_targetWidgetIndex] = {
                ..._targetWidget,
                size: (_targetWidget.size === WIDGET_SIZE.full) ? (widgetSizes[0] ?? WIDGET_SIZE.md) : WIDGET_SIZE.full,
            }
            state.dashboardWidgetInfoList = _dashboardWidgetInfoList;
        }
```